pkg/serving/v1/client.go

```diff
@@ -207,6 +218,37 @@ func (cl *knServingClient) UpdateService(service *servingv1.Service) error {
 	return updateServingGvk(service)
 }
 
+// Update the given service with a retry in case of a conflict
+func (cl *knServingClient) UpdateServiceWithRetry(name string, updateFunc serviceUpdateFunc, nrRetries int) error {
+	return updateServiceWithRetry(cl, name, updateFunc, nrRetries)
+}
+
+// Extracted to be usable with the Mocking client
+func updateServiceWithRetry(cl KnServingClient, name string, updateFunc serviceUpdateFunc, nrRetries int) error {
```

Comment:
should there be more cases to retry other than `apierrors.IsConflict(err)`; we can make that configurable as well, need to check.